apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/Playground/DocumentParams/HistoryLogParams/useLogHistoryParams.ts

```diff
@@ -0,0 +1,136 @@
+import { useCurrentCommit, useCurrentProject } from '@latitude-data/web-ui'
+import { useCurrentDocument } from '$/app/providers/DocumentProvider'
+import {
+  PlaygroundInput,
+  PlaygroundInputs,
+} from '$/hooks/useDocumentParameters'
+import useDocumentLogs from '$/stores/documentLogs'
+import useDocumentLogWithMetadata from '$/stores/documentLogWithMetadata'
+import useDocumentLogsPagination from '$/stores/useDocumentLogsPagination'
+
+import { useSelectedLogRow } from './useSelectedRow'
+
+function getValue({
+  paramValue,
+  inputValue,
+}: {
+  paramValue: unknown | undefined
+  inputValue: PlaygroundInput
+}) {
+  try {
+    return paramValue
+      ? JSON.stringify(paramValue)
+      : typeof inputValue === 'string'
+        ? inputValue
+        : inputValue.value
+  } catch {
+    return typeof inputValue === 'string' ? inputValue : inputValue.value
+  }
+}
+function mapLogParametersToInputs({
+  inputs,
+  parameters,
+}: {
+  inputs: PlaygroundInputs
+  parameters: DocumentLog['parameters'] | undefined
+}): PlaygroundInputs | undefined {
+  const params = parameters ?? {}
+  // No parameters
+  if (!Object.keys(params).length) return undefined
+
+  return Object.entries(inputs).reduce((acc, [key, inputValue]) => {
+    const includedInPrompt = !!params[key]
+    acc[key] = {
+      value: getValue({ paramValue: params[key], inputValue }),
+      includedInPrompt,
+    }
+    return acc
+  }, {} as PlaygroundInputs)
+}
+
+const ONLY_ONE_PAGE = '1'
+export function useLogHistoryParams({
+  inputs,
+  setInputs,
+}: {
+  inputs: PlaygroundInputs
+  setInputs: (newInputs: PlaygroundInputs) => void
+}) {
+  const document = useCurrentDocument()
+  const { saveRowInfo, selectedRow } = useSelectedLogRow({ document })
+  const selectedLogUuid = selectedRow?.documentLogUuid
+  const { project } = useCurrentProject()
+  const { commit } = useCurrentCommit()
+  const [page, setPage] = useState<number>(1)
+  const { data: pagination, isLoading: isLoadingCounter } =
+    useDocumentLogsPagination({
+      projectId: project.id,
+      commitUuid: commit.uuid,
+      documentUuid: document.documentUuid,
+      page: page.toString(),
+      pageSize: ONLY_ONE_PAGE,
+    })
+  const [paginatedLog, setPaginatedLog] = useState<
+    DocumentLogWithMetadataAndError | undefined
+  >()
+  const onLogFetched = useCallback(
+    (logs: DocumentLogWithMetadataAndError[]) => {
+      const log = logs[0]
+      if (!log) return
+      setPaginatedLog(log)
+      saveRowInfo({ documentLogUuid: log.uuid })
+    },
+    [setInputs, saveRowInfo],
+  )
+  const { isLoading } = useDocumentLogs({
+    // NOTE: Nasty hack to avoid fetcing all the logs when
+    // already we have a selected one log UUID in local storage
+    // What we do here is set documentUuid to undefined
+    // so this way the fetcher in this hook will not be called
+    documentUuid: !selectedLogUuid ? document.documentUuid : undefined,
```

Comment:
Any better ideas are welcome. Not happy with this but it works

Reply:
the endpoint is paginated no? What's the harm in letting it fetch the logs?